components/scream/src/diagnostics/tests/potential_temperature_test.cpp

```diff
@@ -0,0 +1,277 @@
+  std::map<std::string,Field> input_fields;
+  for (const auto& req : diag->get_required_field_requests()) {
+    Field f(req.fid);
+    f.allocate_view();
+    const auto name = f.name();
+    f.get_header().get_tracking().update_time_stamp(t0);
+    diag->set_required_field(f.get_const());
+    REQUIRE_THROWS(diag->set_computed_field(f));
+    input_fields.emplace(name,f);
+  }
+
+  // Initialize the diagnostic
+  diag->initialize(t0,RunType::Initial);
+
+  // Run tests
+  const ScalarT p0   = PC::P0;
+  const ScalarT zero = 0.0;
+  const ScalarT one  = 1.0;
+
+  // Get views of input data and set to random values
+  const auto& T_mid_f = input_fields["T_mid"];
+  const auto& T_mid_v = T_mid_f.get_view<ScalarT**>();
+  const auto& p_mid_f = input_fields["p_mid"];
+  const auto& p_mid_v = p_mid_f.get_view<ScalarT**>();
+
+  // Test 1 - property tests 
+  //  - theta(T=0) = 0
+  {
+  Field zero_f = T_mid_f;  // Field with only zeros
+  zero_f.deep_copy(0.0);
+  Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    const auto& T_sub = ekat::subview(T_mid_v,i);
+    const auto& p_sub = ekat::subview(p_mid_v,i);
+    Kokkos::deep_copy(T_sub,zero);
+    Kokkos::deep_copy(p_sub,p0);
+  });
+  Kokkos::fence();
+  const auto& diag_out = diag->get_diagnostic(100.0);
+  REQUIRE(views_are_equal(diag_out,zero_f));
+  }
+  //  - theta=T when p=p0
+  {
+  Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    const auto& T_sub = ekat::subview(T_mid_v,i);
+    const auto& p_sub = ekat::subview(p_mid_v,i);
+    Kokkos::deep_copy(T_sub,temperature);
+    Kokkos::deep_copy(p_sub,p0);
+  });
+  Kokkos::fence();
+  const auto& diag_out = diag->get_diagnostic(100.0);
+  REQUIRE(views_are_equal(diag_out,T_mid_f));
+  }
+  // The output from the diagnostic should match what would happen if we called "calculate_theta_from_T" directly
+  {
+  Field theta_f = T_mid_f;
+  theta_f.deep_copy(0.0);
+  const auto& theta_v = theta_f.get_view<ScalarT**>();
+  Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team,num_mid_packs), [&] (const Int& k) {
+      theta_v(i,k) = PF::calculate_theta_from_T(T_mid_v(i,k),p_mid_v(i,k));
+    });
+    team.team_barrier();
+  });
+  Kokkos::fence();
+  const auto& diag_out = diag->get_diagnostic(100.0);
+  REQUIRE(views_are_equal(diag_out,theta_f));
+  }
+ 
+  // Finalize the diagnostic
+  diag->finalize(); 
+
+} // run()
+
+TEST_CASE("potential_temp_test", "potential_temp_test]"){
+  // Run tests for both Real and Pack, and for (potentially) different pack sizes
+  using scream::Real;
+  using Device = scream::DefaultDevice;
+
+  constexpr int num_runs = 5;
+
+  auto engine = scream::setup_random_test();
+
+  std::cout << " -> Number of randomized runs: " << num_runs << "\n\n";
```

Comment:
You are mixing cout and printf. I'm ok with either one, but let's keep it consistent.

Reply:
this is copy/pasted directly from our common_physics_unit_tests.

Reply:
should I fix there too?